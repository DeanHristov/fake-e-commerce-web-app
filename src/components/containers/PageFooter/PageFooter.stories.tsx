import type { Meta, StoryObj } from '@storybook/react';
import PageFooter, { IPageFooterProps } from './PageFooter';

const meta: Meta<IPageFooterProps> = {
  title: 'Containers/PageFooter',
  component: PageFooter,
  parameters: {},
  argTypes: {},
};

export default meta;

type Story = StoryObj<IPageFooterProps>;

export const Primary: Story = {
  render: () => (
    <PageFooter>
      <h3 className="p-2 text-center text-slate-900">&#169; 2023 | Fake App</h3>
    </PageFooter>
  ),
};
