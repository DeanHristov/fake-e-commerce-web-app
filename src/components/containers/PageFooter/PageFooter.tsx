import { FC, ReactNode } from 'react';
import Divider from '../../ui/Divider';

export interface IPageFooterProps {
  children: ReactNode;
}

const PageFooter: FC<IPageFooterProps> = ({ children }) => {
  return (
    <footer className="p-2">
      <Divider gradient />
      {children}
    </footer>
  );
};

export default PageFooter;
