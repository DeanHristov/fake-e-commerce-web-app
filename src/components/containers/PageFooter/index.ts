import PageFooter, { IPageFooterProps } from './PageFooter';

export type { IPageFooterProps };

export default PageFooter;
