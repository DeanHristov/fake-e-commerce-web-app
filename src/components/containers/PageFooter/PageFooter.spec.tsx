import { render } from '@testing-library/react';
import PageFooter from './PageFooter';

describe('Containers/PageFooter <PageFooter {...} />', () => {
  it('Should render default state', () => {
    const { container } = render(
      <PageFooter>
        <h3 className="p-2 text-center text-slate-900">
          &#169; 2023 | Fake App
        </h3>
      </PageFooter>,
    );

    expect(container).toMatchSnapshot();
  });
});
