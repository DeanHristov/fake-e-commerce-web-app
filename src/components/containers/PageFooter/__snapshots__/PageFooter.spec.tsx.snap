// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Containers/PageFooter <PageFooter {...} /> Should render default state 1`] = `
<div>
  <footer
    class="p-2"
  >
    <div
      class="w-full h-px m-auto bg-gradient-to-r from-transparent via-gray-400 to-transparent"
    />
    <h3
      class="p-2 text-center text-slate-900"
    >
      Â© 2023 | Fake App
    </h3>
  </footer>
</div>
`;
